### DISCUSSION TITLE
Need help with NextJS + Electron

### DISCUSSION THREAD

**Discussion by alancunha26** [vercel/vercel#5241] [Help] [answered]
### The problem

Hello guys, I have an application that will be desktop supported (using native modules) and will be published in web as well. The main problem is that I would like to have all the befits of using next for developing the project, because web would be SEO friendly with all the revalidation stuff, but my app need to run in Electron too, because I need access to all desktop tools and apis that Electron provides.

I made a lot of research about this matter, and find some solutions, but nothing that looks really production friendly. Maybe I am being too picky and should leave next and use plain react, but I really want to my web application be server side rendered. Maybe someone here could help me with this or give me another solutions.

Thanks for your help and sorry for my english!

### Features needed:
- Shared code between web and desktop;
- Server side render for web (SEO);
- Export native modules and features from electron to renderer;

### References
- https://github.com/saltyshiomix/nextron
- https://github.com/vercel/next.js/tree/master/examples/with-electron-typescript
- https://leo.im/2017/electron-next

**Comment by markozxuu** [comment, marked as answer]
Hi @alancunha26  👋  Your question is more focused on Next.js, so I invite you to ask future questions in the [community](https://github.com/vercel/next.js/discussions) they have on GitHub.  Answering your question, there is no current thing that meets your particular requirements. Next.js has changed a lot since the `electron-next` project was launched.

**Reply by alancunha26** [reply]
Hello @markozxuu thanks for your quick reply, I will do as you recommended.

**Comment by alancunha26** [comment]
If someone would like to see the discussion as it goes, it is [here](https://github.com/vercel/next.js/discussions/17543) now.
